src/send-email.js

```diff
@@ -3,52 +3,118 @@ require('dotenv').config()
+  const emails = await fetchEmailsFromRepo(repoOwner, repoName, accessToken)
+}
 
 
-}
+async function fetchEmailsFromRepo(repoOwner, repoName, accessToken) {
 
-async function fetchFilesFromRepo() {
-  const repoOwner = process.env.repoOwner
-  const repoName = process.env.repoName
-  const folderPath = 'content/project'
-  const accessToken = process.env.PAT
-  const apiUrl = `https://api.github.com/repos/${repoOwner}/${repoName}/contents/${folderPath}?ref=add-example-projects&media=raw`
+  const getEmailsPath = 'content/config/email-service-contacts.json';
+  const apiUrl = `https://api.github.com/repos/${repoOwner}/${repoName}/contents`
 
   try {
-    const response = await axios.get(apiUrl, {
+
+    const emailsUrl = `${apiUrl}/${getEmailsPath}?ref=add-example-projects&media=raw`;
+    const response = await axios.get(emailsUrl, {
       headers: {
         Authorization: `Bearer ${accessToken}`,
         Accept: 'application/vnd.github.v3+json'
       }
     })
 
-    if (Array.isArray(response.data)) {
-      const fileContent = response.data.map(file => {
+    const downloadEmailsUrl = response.data.download_url;
+    const downloadEmailsResponse = await axios.get(downloadEmailsUrl);
+    return downloadEmailsResponse.data.contacts
+
+  } catch (error) {
+    console.error('Error with fetchEmailsFromRepo():', error)
+  }
+  return []
+}
+
+async function fetchFilesFromRepo(repoOwner, repoName, accessToken) {
+
+  const getProjectsPath = 'content/project'
+  const getContactsPath = 'content/contact';
+  const apiUrl = `https://api.github.com/repos/${repoOwner}/${repoName}/contents`
+
+  try {
+
+    const projectsUrl = `${apiUrl}/${getProjectsPath}?ref=add-example-projects&media=raw`
+    const contactsUrl = `${apiUrl}/${getContactsPath}?ref=add-example-projects&media=raw`;
+
+    const [projectsResponse, contactsResponse] = await Promise.all([
+      axios.get(projectsUrl, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+          Accept: 'application/vnd.github.v3+json'
+        }
+      }),
+      axios.get(contactsUrl, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+          Accept: 'application/vnd.github.v3+json'
+        }
+      })
+    ])
+
+
+    if (Array.isArray(contactsResponse.data) && Array.isArray(projectsResponse.data)) {
+
+      const contactsContent = contactsResponse.data
+        .filter(file => file.name.endsWith('.json'))
+        .map(file => {
+          const fileUrl = file.download_url;
+          return axios.get(fileUrl).then(response => response.data);
+        });
+
+      const contactsContentArray = await Promise.all(contactsContent);
+
+      const contactsMap = new Map()
+      let index = 0
+
+      contactsResponse.data.forEach((file) => {
+        if (file.name.endsWith('.json')) {
+          const fileName = file.name;
+          const fileData = contactsContentArray[index++];
```

Comment:
The reason I did it this funky way is because i need to grab data from 2 places. 

This is how the requests/responses work with axios. 

Our first request, captured here

```
const [projectsResponse, contactsResponse] = await Promise.all([
      axios.get(projectsUrl, {
        headers: {
          Authorization: `Bearer ${accessToken}`,
          Accept: 'application/vnd.github.v3+json'
        }
      }),
      axios.get(contactsUrl, {
        headers: {
          Authorization: `Bearer ${accessToken}`,
          Accept: 'application/vnd.github.v3+json'
        }
      })
    ])
```

returns the following in contactsResponse

```
response object = {
 ...
 data: [
    {
      name: 'brown-logo_2016_2-color-process-hz_2400.png',
      path: 'content/contact/brown-logo_2016_2-color-process-hz_2400.png',
      sha: '8f25b24f941c04c0397b93c5c48ae3185cba2211',
      size: 28608,
      url: 'https://api.github.com/repos/thepolicylab-projectportals/example-content/contents/content/contact/brown-logo_2016_2-color-process-hz_2400.png?ref=main',
      html_url: 'https://github.com/thepolicylab-projectportals/example-content/blob/main/content/contact/brown-logo_2016_2-color-process-hz_2400.png',
      git_url: 'https://api.github.com/repos/thepolicylab-projectportals/example-content/git/blobs/8f25b24f941c04c0397b93c5c48ae3185cba2211',
      download_url: 'https://raw.githubusercontent.com/thepolicylab-projectportals/example-content/main/content/contact/brown-logo_2016_2-color-process-hz_2400.png?token=AIECF5JHULW7POUPJGWGWYLEYMQLM',
      type: 'file',
      _links: [Object]
    },
    {
      name: 'center-for-computation-and-visualization.json',
      path: 'content/contact/center-for-computation-and-visualization.json',
      sha: '54cdf503a8a3310ae24cf6f57d0a84b3aebd8a59',
      size: 194,
      url: 'https://api.github.com/repos/thepolicylab-projectportals/example-content/contents/content/contact/center-for-computation-and-visualization.json?ref=main',
      html_url: 'https://github.com/thepolicylab-projectportals/example-content/blob/main/content/contact/center-for-computation-and-visualization.json',
      git_url: 'https://api.github.com/repos/thepolicylab-projectportals/example-content/git/blobs/54cdf503a8a3310ae24cf6f57d0a84b3aebd8a59',
      download_url: 'https://raw.githubusercontent.com/thepolicylab-projectportals/example-content/main/content/contact/center-for-computation-and-visualization.json?token=AIECF5LCDROYCNEWDHHZBOLEYMQLM',
      type: 'file',
      _links: [Object]
    },
    {
      name: 'test-contact.json',
      path: 'content/contact/test-contact.json',
      sha: '7332407f3729f5352682f523b39e2c1e0cfc3ea8',
      size: 164,
      url: 'https://api.github.com/repos/thepolicylab-projectportals/example-content/contents/content/contact/test-contact.json?ref=main',
      html_url: 'https://github.com/thepolicylab-projectportals/example-content/blob/main/content/contact/test-contact.json',
      git_url: 'https://api.github.com/repos/thepolicylab-projectportals/example-content/git/blobs/7332407f3729f5352682f523b39e2c1e0cfc3ea8',
      download_url: 'https://raw.githubusercontent.com/thepolicylab-projectportals/example-content/main/content/contact/test-contact.json?token=AIECF5IDVK5HUGWMLXPY4ETEYMQLM',
      type: 'file',
      _links: [Object]
    }
  ]
}
```

So I have to go and use the download_url to further request the actual content for each contact. But because heather is requesting the slug, the only way i can obtain this is from the name of the file, which only exists in this first response object. So when creating the mapping, i am attempting to grab data from both response objects, and the reason for using the index is because some of the data is the png files for the logos. So thats why i need a way to skip over these files by using a counter to keep track of the index